### PR TITLE
Refactoring DX

### LINKED ISSUE #108: Error when crawling history
```
----------------------------------------
-- CRAWL PHASE ---------------------------------------------------
� Looking for deprecations in:  6.2.0
Date: 2018-05-21T21:48:49-07:00
--- ⏰
✓  Found  125  in  201  files.

----------------------------------------
(node:11228) UnhandledPromiseRejectionWarning: Error: Previous HEAD position was 7778d810 chore(publish): 6.2.0
fatal: update_ref failed for ref 'HEAD': couldn't set 'HEAD'

    at GitExecutorChain.onFatalException (C:\Users\micha\git\find-deprecations\node_modules\simple-git\src\lib\runners\git-executor-chai
n.js:60:87)
    at GitExecutorChain.<anonymous> (C:\Users\micha\git\find-deprecations\node_modules\simple-git\src\lib\runners\git-executor-chain.js:
51:28)
    at Generator.throw (<anonymous>)
    at rejected (C:\Users\micha\git\find-deprecations\node_modules\simple-git\src\lib\runners\git-executor-chain.js:6:65)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:11228) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async
function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandle
d promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode
). (rejection id: 2)
(node:11228) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not h
andled will terminate the Node.js process with a non-zero exit code.

```

### PR DESCRIPTION
**Pull request by BioPhoton**
This PR includes a lot of small changes. I tried to have them in separate commits.

Please check them for sanity ;)

- Fix order in json
- Removed commit task
- Moved question for grouping into commands
- Refine RxJS example
- Add stash changes before checkout next version when walking the tag history
- insert command to format raw-json as command

closes #108

### CODE REVIEW

File: packages/deprecation-crawler/src/lib/utils.ts [hunk 1]
```diff
@@ -122,15 +126,45 @@ export function readRawDeprecations(config: CrawlConfig) {
   return { deprecations, path };
 }
 
+export function writeRawDeprecationsTask(config: CrawlConfig): CrawlerProcess {
```
Comment (timdeschryver) [inline packages/deprecation-crawler/src/lib/utils.ts:129]: Should this be moved to the tasks?

File: packages/deprecation-crawler/src/lib/commands/history.command.ts [hunk 2]
```diff
@@ -60,7 +60,7 @@ export const historyCommand: YargsCommandObject = {
 
       const deprecations: Deprecation[] = [];
       for (const tag of tagsToCrawl) {
-        const tasks = [checkout, crawl, addVersion, commitChanges];
+        const tasks = [checkout, crawl, addVersion, ensureCleanGit];
```
Comment (timdeschryver) [inline packages/deprecation-crawler/src/lib/commands/history.command.ts:63]: Why are we using `ensureCleanGit` here? 
This stashes the changes, so the user would have to unstash them and then commit the changes?

**Review by timdeschryver — COMMENTED** [review]
Can you take a look at the failing test?

File: packages/deprecation-crawler/src/lib/commands/history.command.ts (hunk 2 above)
Comment (BioPhoton) [reply packages/deprecation-crawler/src/lib/commands/history.command.ts:63]: It was a way it worked for me to crawle/walkthrough  a changing file structure.

I should have use revert imho. I guess i didn't though too much. Plus I was not 100% why the walkthrough broke... 

I'll have a proper think on Saturday latest ✌️

File: packages/deprecation-crawler/src/lib/utils.ts (hunk 1 above)
Comment (BioPhoton) [reply packages/deprecation-crawler/src/lib/utils.ts:130]: Nice catch! Agree.

Comment (BioPhoton) [reply packages/deprecation-crawler/src/lib/utils.ts:130]: Moved to tasks

File: packages/deprecation-crawler/src/lib/commands/history.command.ts (hunk 2 above)
Comment (BioPhoton) [reply packages/deprecation-crawler/src/lib/commands/history.command.ts:63]: I use `git clean -f` now

**Comment by BioPhoton** [comment]
done!

**Pull request merged**